osu.Game/Screens/Select/Details/BeatmapDetailsGraph.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Game.Screens.Select.Details
+{
+    public class BeatmapDetailsGraph : FillFlowContainer<BeatmapDetailsBar>

```

Comment:
- make this into a generic `BarGraph` component with its own visualtest.
- rather than bothering with this `BarDirection` thing, why not just use the `FillFlow`'s `Direction`? you could rotate the bars and then the `Bar` class (which should be nested inside `BarGraph`) would need to knowledge about its direction.